com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -1436,18 +1437,15 @@ private void HandleRawTransportPoll(NetworkEvent networkEvent, ulong clientId, A
 #if DEVELOPMENT_BUILD || UNITY_EDITOR
                     s_TransportDisconnect.Begin();
 #endif
-                    clientId = TransportIdToClientId(clientId);
-
-                    OnClientDisconnectCallback?.Invoke(clientId);
-
-                    m_TransportIdToClientIdMap.Remove(transportId);
-                    m_ClientIdToTransportIdMap.Remove(clientId);
+                    clientId = TransportIdCleanUp(clientId, transportId);
 
                     if (NetworkLog.CurrentLogLevel <= LogLevel.Developer)
                     {
                         NetworkLog.LogInfo($"Disconnect Event From {clientId}");
                     }
 
+                    OnClientDisconnectCallback?.Invoke(clientId);
```

Comment:
For the client that never connected, here, we will have skipped `TransportIdToClientId`. As such, we'll end up calling `OnClientDisconnectCallback` with a transport id? It feels kind-of like we're passing a parameter that is not in the right range of values.

But, on the flip side, if the client never connected, it's not like it could have a valid client id, so I guess it's OK. I just wonder if the client id is correct in that case. see below.